### ISSUE TITLE
Show warning if pdf does not match expectations

### ISSUE THREAD

**Issue by pietermarsman** [pdfminer/pdfminer.six#432]
**Feature request**

Pdfminer.six can handle some irregularities in a pdf. This is nice because often PDF's do not follow the standard. This might cause unexpected results, and people might attribute these to pdfminer.six while in fact the PDF is the root of the error. 

By showing a warning we make users aware that the PDF itself might be the root problem. Also, when people ask for help, this gives us a better understanding of what could be fixed. 

To implement this, we can start with finding all usages of `setttings.STRICT` and raise a warning if it is `False`. This might even be a util function that we can call with an error. 

Examples of errorneous pdfs:

- #193

**Comment by jstockwin** [comment]
This is a good idea. Keeping pdfminer.six robust is very helpful and important, but when we do our best to fix an issue with the PDF, it would be good to be told what is happening. It should be easily turned-off, though.

It might be helpful to be able to turn off certain warnings, too? I.e. if I know I have problem X in my PDF, then I might want to run in strict mode excluding problem X, so that if my PDF suddenly develops problem Y an exception is raised.

**Comment by pietermarsman** [comment]
I think I remember the warnings module can be used to silent specific (or all?) warnings.

Extending strict mode to be more configurable, e.g. allowing one problem while raising an error for another, could be another issue.

**Comment by jstockwin** [comment]
Yeah, I agree extending strict mode is probably another issue, and maybe not that important.

I think a util function (something like `error_or_warning`) would work well here.

Having just searched the project for `settings.STRICT` there are quite a few hits. It's generally followed by possibly raising an exception, and most of the exceptions have reasonable messages attached. That said, when throwing a warning, it would be nice to say what we're (trying?) to do to fix it.

E.g. currently we do `raise XYZException("Foo is wrong.")`. It would be nice if the warning was `logger.warning("Foo is wrong, trying to fix by doing Bar")`. This will need a good understanding of the internals, though, so could be too much work? (Certainly, I wouldn't have a clue what the results of ignoring the exception are in a lot of places).

**Comment by pietermarsman** [comment]
I agree that showing both the problem, and the thing that we are trying to do about is even better.

I don't think it's a lot of work because the code execution doesn't change. Errors are still raised if strict mode is on, and if its off only an additional warning is shown. So basically, we need to add a lot of warnings in the places were `settings.STRICT` is used.
